vignettes/workflows.Rmd

```diff
@@ -0,0 +1,210 @@
+
+## Group of researchers, sharing via server
+
+This is the model that we use within the [MRC Centre for Global Infectious Disease Analysis](https://www.imperial.ac.uk/mrc-global-infectious-disease-analysis), and it does require considerably more setup, but also more reliability and flexibility.
+
+As above, we use a single "source tree", kept on GitHub or similar.  However, someone in the group takes responsibility for setting up a copy of [OrderlyWeb](https://github.com/vimc/orderly-web), our open source, self-hosted orderly server.  The OrderlyWeb server software provides a number of things
+
+1. A persistent copy of R running `orderly` that can run reports as needed
+2. A web server that exposes a user-friendly web portal showing versions of orderly reports and an HTTP API (this application is written in [Kotlin](https://kotlinlang.org))
+3. A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) used to secure the application with https
+
+This [diagram may help show how the pieces fit together](https://raw.githubusercontent.com/vimc/montagu-machine/master/docs/diagrams/OrderlyWeb%20architecture.png).
+
+If OrderlyWeb has been deployed onto `orderly.example.com`, then we might declare it in `orderly_config.yml`
+
+```yaml
+remote:
+  default:
+    driver: orderlyweb::orderlyweb_remote
+    args:
+      host: orderly.example.com
+      token: $GITHUB_TOKEN
+```
+
+Once this is done, both Alice and Bob can run reports *on* that server, for example
+
+```r
+orderly::orderly_run_remote("report-a", remote = "default")
+```
+
+**ADD A GIF**
+
+this will print to screen the log, just as if it was running locally, though it is in fact running on a fresh R session remotely.  After being run, this report is available to everyone and can be pulled as above
+
+```r
+orderly::orderly_pull_archive("report-a", remote = "default")
+```
+
+## Using staging environments
+
+This can be extended further with the use of "staging environments".  This extends the above (two) scenario so that we have multiple remote orderly archives.  These might be shared file archives, using `orderly::orderly_remote_path` or server hosted archives using `orderlyweb::orderlyweb_remote` or even a combination.
+
+We designate one of these remotes to be our production environment and the others to be staging.  In our work we have used the name pairs `production/science` and `real/testing`; the names used do not matter.
+
+The `orderly_config.yml` configuration might then look like:
+
+```yaml
+remote:
+  staging:
+    driver: orderlyweb::orderlyweb_remote
+    args:
+      host: orderly.example.com
+      port: 11443
+      token: $GITHUB_TOKEN
+  production:
+    driver: orderlyweb::orderlyweb_remote
+    args:
+      host: orderly.example.com
+      token: $GITHUB_TOKEN
+```
+
+Now, as part of the pull request review procedure, the reviewer will run the report on the staging environment, which ensures that it will likely work without error on the production environment.
```

Comment:
Something about the words "ensures" and "likely" being close together...
perhaps "should ensure that it will work without error" ?